tests/pausing_test.go

```diff
@@ -363,30 +364,32 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 	Context("A long running process", func() {
 
+		grepSleepPid := func(expecter expect.Expecter) string {
+			res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
+				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
+				&expect.BExp{R: tests.Retcode("[0-9]+", "\\# ")}, // pid
+			}, 15*time.Second)
+			log.DefaultLogger().Infof("a:%+v\n", res)
+			Expect(err).ToNot(HaveOccurred())
+			re := regexp.MustCompile("\r\n[0-9]+\r\n")
+			return strings.TrimSpace(re.FindString(res[0].Match[0]))
+		}
+
 		startProcess := func(expecter expect.Expecter) string {
 			By("Start a long running process")
-			res, err := expecter.ExpectBatch([]expect.Batcher{
+			res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
 				&expect.BSnd{S: "sleep 5&\n"},
 				&expect.BExp{R: "\\# "}, // prompt
-				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
-				&expect.BExp{R: "sleep 5"},    // command
-				&expect.BExp{R: "[0-9]+\r\n"}, // pid
 			}, 15*time.Second)
 			log.DefaultLogger().Infof("a:%+v\n", res)
 			Expect(err).ToNot(HaveOccurred())
-			return strings.TrimSpace(res[2].Match[0])
+
+			return grepSleepPid(expecter)
```

Comment:
Makes sense to call `checkProcess`. See next comment below.

Reply:
I think that `grepSleepPid` explains better what this method is doing. It doesn't check a generic process, it greps the sleep (not any sleep, the `sleep 5` that we just sent).